tests/integration/helpers.py

```diff
@@ -186,21 +229,36 @@ def get_application_unit_ids_ips(ops_test: OpsTest, app: str = APP_NAME) -> Dict
         Dictionary unit_id / unit_ip, of the application
     """
     result = {}
+    for unit in await get_application_units(ops_test, app):
+        result[unit.id] = unit.ip
+
+    return result
+
+
+async def get_application_unit_ids_hostnames(
+    ops_test: OpsTest, app: str = APP_NAME
+) -> Dict[int, str]:
+    """List the units of an application by id and corresponding host name."""
+    result = {}
     for unit in ops_test.model.applications[app].units:
-        result[int(unit.name.split("/")[1])] = unit.public_address
+        unit_id = int(unit.name.split("/")[1])
+        _, raw_hostname, _ = await ops_test.juju("ssh", f"{app}/{unit_id}", "hostname")
```

Comment:
I don't see where we use `raw_hostname`. I think this line is replaced by the next line